airbyte-db/lib/configs_db_schema.md

```diff
@@ -0,0 +1,361 @@
+  - Foreign key columns are named as `<parent_table_name>_id`. They are always indexed.
+  - If the referenced table is polymorphic, the foreign key is broken down into two columns: `owner_type` and `owner_id`.
+- For complex properties, the properties will be expanded to its own table.
+  - The property table is named as `<parent_table_name>_<property_name>`. For example, `WORKSPACE_NOTIFICATION` specifies the notifications for a `WORKSPACE`.
+- Use Table-per-Type to implement table inheritance.
+  - Comparison with other approaches: https://stackoverflow.com/a/3579462
+  - There is a `<base_table_name>_type` column in the base table so that it is easy to know which child table to query. This column is indexed.
+  - The child table has 1:1 mapping with the base table. This is enforced by a unique index on the foreign key in the child table that references the base table.
+- Each table has two timestamp columns: `created_at` and `updated_at`.
+- Index
+  - Single column index is marked with `index`.
+  - Composite column index is marked with `composite index` and specified after column definition.
+
+## Open Questions
+- Should we keep the `standard` prefix in the table names?
+  - Currently, it is removed.
+  - Will we ever need non-standard models like custom workspace or custom connector definition?
+- Most source and destination tables have exactly the same columns. For example, `SOURCE_DEFINITION` and `DESTINATION_DEFINITION`, `SOURCE_OAUTH_PARAM` and `DESTINATION_OAUTH_PARAM`. Should we merge these tables, and add a `type` column to distinguish the rows?
+  - Things to consider
+    - Dry-ness: one table is better
+    - Query complexity: same
+    - Add new columns
+      - Separate tables: need to add new columns to both tables.
+      - One table: add new columns to one table.
+      - One table is better.
+  - Do we expect the source and definition models to diverge in the future?
+    - If we use one table for both source and definition models, and they diverge in the future, we can use the table-per-type strategy and simply add extra tables for the model that needs extra properties. Still one table is better
+  - Based on the above dimensions, I am leaning towards merging the two tables.
+
+## Common Columns
+| Column | Type  | Required | Default | Notes |
+| ------ | :---: | :---:    | :---:   | ---   |
+| id | char(36) | yes | | Primary key, UUID. |
+| created_at | timestamp with time zone | yes | current_timestamp | `current_timestamp` is implemented in most databases. |
+| updated_at | timestamp with time zone | yes | current_timestamp | `current_timestamp` is implemented in most databases. |
+
+## Tables
+
+### `STANDARD_WORKSPACE`
+
+YAML [reference](https://github.com/airbytehq/airbyte/blob/master/airbyte-config/models/src/main/resources/types/StandardWorkspace.yaml)
+
+| Column | Type  | Required | Default | Notes |
```

Comment:
`workspaceId` seems to be missing

Reply:
`workspaceId` is the primary id for this model. So it is replaced with the `id` column specified in the common columns section. In this way we don't need to store or remember which field is the id field.